catalog/transformer/catalog-transformer-microsoft/src/main/java/ddf/catalog/transformer/input/MicrosoftTransformer.java

```diff
@@ -0,0 +1,296 @@
+
+public class MicrosoftTransformer implements InputTransformer {
+    private static final Logger LOGGER = LoggerFactory.getLogger(MicrosoftTransformer.class);
+
+    private Templates templates = null;
+
+    private ClavinFacade clavinFacade;
+
+    public MicrosoftTransformer(BundleContext bundleContext, ClavinFacade clavinFacade) {
+        this.clavinFacade = clavinFacade;
+
+        try {
+            templates = new TransformerFactoryImpl().newTemplates(new StreamSource(
+                    TikaMetadataExtractor.class.getResourceAsStream("/metadata.xslt")));
+        } catch (TransformerConfigurationException e) {
+            LOGGER.warn("Couldn't create XML transformer", e);
+        }
+
+        if (bundleContext == null) {
+            LOGGER.error("Bundle context is null. Unable to register {} as an osgi service.",
+                    MicrosoftTransformer.class.getSimpleName());
+            return;
+        }
+
+        registerService(bundleContext);
+        IIORegistry.getDefaultInstance().registerServiceProvider(new J2KImageReaderSpi());
+        IIORegistry.getDefaultInstance().registerServiceProvider(new TIFFImageReaderSpi());
+    }
+
+    @Override
+    public Metacard transform(InputStream input) throws IOException, CatalogTransformerException {
+        return transform(input, null);
+    }
+
+    @Override
+    public Metacard transform(InputStream input, String id)
+            throws IOException, CatalogTransformerException {
+        LOGGER.debug("Transforming input stream using Tika.");
+
+        if (input == null) {
+            throw new CatalogTransformerException("Cannot transform null input.");
+        }
+
+        try (FileBackedOutputStream fileBackedOutputStream = new FileBackedOutputStream(1000000)) {
+            try {
+                IOUtils.copy(input, fileBackedOutputStream);
+            } catch (IOException e) {
+                throw new CatalogTransformerException("Could not copy bytes of content message.",
+                        e);
+            }
+
+            Parser parser = new AutoDetectParser();
+            ToXMLContentHandler handler = new EmptyBodyXMLHandler();
+            TikaMetadataExtractor tikaMetadataExtractor = new TikaMetadataExtractor(parser,
+                    handler);
+
+            Metadata metadata;
+            try (InputStream inputStreamCopy = fileBackedOutputStream.asByteSource().openStream()) {
+                metadata = tikaMetadataExtractor.parseMetadata(inputStreamCopy, new ParseContext());
+            }
+
+            String metadataText = handler.toString();
+            if (templates != null) {
+                metadataText = transformToXml(metadataText);
+            }
+
+            Metacard metacard = MetacardCreator.createBasicMetacard(metadata, id, metadataText);
+
+            try (InputStream is = fileBackedOutputStream.asByteSource().openStream()) {
+                try {
+                    String document = IOUtils.toString(is, "UTF-8");
```

Comment:
This can't be in here and will blow up the heap. Do not use IOUtils.toString or IOUtils.toByteArray